### PR TITLE
Io mini1800 - New Keyboard

### PR DESCRIPTION
**Pull request by jpuerto96**
## Description

This PR adds a new keyboard to QMK - IO-Mini 1800

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/io_mini1800/rules.mk [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (fauxpark) [inline keyboards/io_mini1800/rules.mk:15]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/io_mini1800/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/io_mini1800/keymaps/default/keymap.c:31]:
```suggestion
```

File: keyboards/io_mini1800/keymaps/2x3u/keymap.c [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+    return true;
+}
+
+
+
```
Comment (drashna) [inline keyboards/io_mini1800/keymaps/2x3u/keymap.c:34]:
```suggestion
```

File: keyboards/io_mini1800/io_mini1800.c [hunk 4]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2022 s8erdude (@jpuerto96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "io_mini1800.h"
```
Comment (drashna) [inline keyboards/io_mini1800/io_mini1800.c:4]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "io_mini1800.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by jpuerto96** [comment]
@drashna ping for re-review.

**Comment by jpuerto96** [comment]
@drashna - re-pinging for re-review.

File: keyboards/io_mini1800/readme.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+    make io_mini1800:default
+
+Flashing example for this keyboard:
+
+    make io_mini1800:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+To reset the board, you will want to short the pad labeled RESET on the back (near the MCU).
```
Comment (noroadsleft) [inline keyboards/io_mini1800/readme.md:19]:
```suggestion
## Bootloader

To reset the board into bootloader, do one of the following:

* **Hardware reset**: short the pad labeled RESET on the back, near the microcontroller
* **Bootmagic reset**: hold down the top left key (usually programmed as Escape) while plugging in the keyboard (also resets persistent storage)
```

File: keyboards/io_mini1800/keymaps/2x3u/readme.md [hunk 6]
```diff
@@ -0,0 +1 @@
+# The default keymap for io_mini1800
```
Comment (noroadsleft) [inline keyboards/io_mini1800/keymaps/2x3u/readme.md:1]:
```suggestion
# The default Split Spacebar keymap for io_mini1800
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**